docs/operating-eck/restrict-cross-namespace-associations.asciidoc

```diff
@@ -0,0 +1,94 @@
+
+This is true, for example, when using the `elasticsearchRef` field to automatically configure and establish a trust relationship between Kibana and Elasticsearch. If a user can create a Kibana instance in a namespace managed by ECK, then this Kibana instance can be associated with any Elasticsearch cluster deployed in another namespace managed by the same ECK instance, even if the user only has permissions to edit Kibana and not the Elasticsearch resource.
+
+This section describes how you can restrict the associations that can be created between resources managed by ECK.
+
+== Enabling cross namespace association restriction
+
+This feature is disabled by default. To enable it, start the operator with the `--enforce-rbac-on-refs` flag.
+
+NOTE: This feature only provides an access control for resources deployed across two different namespaces. Associations can still be freely created between resources deployed in the same namespace.
+
+Once enabled, associations can be created as long as the `ServiceAccount` specified in the associated resources can execute an HTTP `get` request against the referenced Elasticsearch object.
+
+IMPORTANT: ECK removes any association that is not allowed. If you have existing associations, do not enable this feature without creating the required `Roles` and `RoleBindings` as described in the following sections.
+
+== Allowing associations to be created
+
+First create a `ServiceAccount`:
+
+[source,sh]
+----
+> kubectl create serviceaccount associated-resource-sa
+----
+
+Then set the `serviceAccountName` field in the associated resource to specify which `ServiceAccount` is used to create the association:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: <kind>.k8s.elastic.co/{eck_crd_version}
+kind: <Kind>
+metadata:
+  name: associated-resource
+  namespace: associated-resource-ns
+spec:
+ ...
+  elasticsearchRef:
+    name: "elasticsearch-sample"
+    namespace: "elasticsearch-ns"
+  # Service account used by this resource to get access to an Elasticsearch cluster
+  serviceAccountName: associated-resource-sa
+----
+
+In the above example, `associated-resource` can be of any `Kind` that requires an association to be created (e.g. `Kibana` or the `APMServer`).
+You can find https://github.com/elastic/cloud-on-k8s/blob/master/config/recipes/associations-rbac/apm_es_kibana_rbac.yaml[a complete example in the ECK GitHub repository].
+
+NOTE: If the `serviceAccountName` is not set then ECK uses the `ServiceAccount` called `default`
+
+Create a `ClusterRole` to allow an HTTP `get` request to be run against Elasticsearch objects:
+
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: elasticsearch-association
+rules:
+  - apiGroups:
+      - elasticsearch.k8s.elastic.co
+    resources:
+      - elasticsearches
+    verbs:
+      - get
+----
+
+Then, to allow the `ServiceAccount` of the associated resource to use this role create the following `RoleBinding`:
```

Comment:
```suggestion
Then, create a `RoleBinding` in the Elasticsearch namespace to allow any resource presenting the correct service account token to associate with the Elasticsearch cluster:
```

Reply:
I wonder if "presenting the correct service account token" is overly specific. Maybe just:
>Then, create a `RoleBinding` in the Elasticsearch namespace to allow any resource using the `ServiceAccount` to associate with the Elasticsearch cluster

?